PythonPonies/bangazon-storefront
# Ike add products refactor

# Description
Add description about what this pull request covers.
Refactor adding a product feature to the product owner's liking.

## Number of Fixes
1

## Related Ticket(s)
List related ticket(s) that apply to the work done in this pull request
Add product to product list
## Problem to Solve
Describe the problem(s) this pull request solves.
Refactors the functionality of the add product feature.
## Proposed Changes
Describe the proposed changes

Selected a product type by dropdown OR Add a product type by input.
THEN, add a product to sell. 

## Expected Behavior
Describe the expected behavior.
Add a category, select any category including the one just added. Add a product to the selected category. 
## Steps to Test Solution
git fetch branch
delete database and migrate DB
Run the server
Navigate the site to add products section
Add a category
Select a category
Add a product
check to see product is included in the database

## Testing

[ ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[x ] I certify that all existing tests pass

## Documentation
[ ] There is new documentation in this pull request that must be reviewed..
[ x] I added documentation for any new classes/methods

## Deploy Notes
Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.
